### PR TITLE
Pepper Spray update

### PR DESCRIPTION
**Pull request by HMBGERDO**
## What Does This PR Do

This PR changes pepper spray reagent effects

If it hits target with eyes not covered, it will do those effects:
0.5 seconds stun
20 seconds blurry eye
8 seconds blind

If it hits target with no mask cover, it will apply refreshable status effect with those effects:
lasts 10 seconds

on_apply(first hit):
12 seconds confused
4 seconds slowed
40 stamina damage

on_refresh(second and more hits until status effect is applied):
12 seconds confused
4 seconds slowed
20 stamina damage

## Why It's Good For The Game

Currently pepper spray is either completely useless or completely OP weapon. You either stunlock target in first hit or cant do anything to them. This design is bad. 
Im trying to make it useful in most situations here, so those are changes.
Discussed those with two balance guys.

UPD: just making it less painful for not protected targets, removing hardstun for 6 seconds

## Testing

Compiled, used pepper spray on self, on other man, checked that it works as intended

## Changelog
:cl:
tweak: Pepper spray hitting eyes now stuns for 0.5 seconds, blinds for 8 seconds and blurs for 20 seconds. If target has no mask, that will do 40 stamina damage, 4 seconds slowed and 12 seconds confused. Hitting one target twice will make second stamina damage 20 instead of 40.
/:cl:

### CODE REVIEW

File: code/datums/status_effects/debuffs.dm [hunk 1]
```diff
@@ -382,6 +382,26 @@
...
+/datum/status_effect/pepper_spray
+	id = "pepperspray"
+	duration = 10 SECONDS
+	status_type = STATUS_EFFECT_REFRESH
+	tick_interval = -1
+	alert_type = null
+
+/datum/status_effect/pepper_spray/on_apply()
+	. = ..()
+	to_chat(owner, "<span class='danger'>You inhale pepper spray!</span>")
```
Comment (DGamerL) [inline code/datums/status_effects/debuffs.dm:394]:
```suggestion
	to_chat(owner, "<span class='danger'>Your throat burns!</span>")
```
Seems a bit more what you'd feel, in my opinion

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 2]
```diff
@@ -219,56 +219,31 @@
 	if(method == REAGENT_TOUCH)
 		if(ishuman(M))
 			var/mob/living/carbon/human/victim = M
-			var/mouth_covered = 0
 			var/eyes_covered = 0
```
Comment (DGamerL) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm:222]:
```suggestion
			var/eyes_covered = FALSE
```

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 3]
```diff
@@ -219,56 +219,31 @@
...
-					mouth_covered = 1
-					safe_thing = victim.head
+			if(victim.wear_mask && victim.wear_mask.flags_cover & MASKCOVERSEYES)
+				eyes_covered = 1
+				safe_thing = victim.wear_mask
+			if(victim.head && victim.head.flags_cover & MASKCOVERSEYES)
+				eyes_covered = 1
+				safe_thing = victim.head
 			if(victim.glasses)
 				eyes_covered = 1
```
Comment (DGamerL) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm:231]:
```suggestion
				eyes_covered = TRUE
				safe_thing = victim.wear_mask
			if(victim.head && victim.head.flags_cover & MASKCOVERSEYES)
				eyes_covered = TRUE
				safe_thing = victim.head
			if(victim.glasses)
				eyes_covered = TRUE
```

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 4]
```diff
@@ -219,56 +219,31 @@
...
-				victim.Confused(12 SECONDS)
-				victim.damageoverlaytemp = 75
-				victim.Weaken(10 SECONDS)
-				victim.drop_item()
+
+			if(!victim.internal)
+				victim.apply_status_effect(STATUS_EFFECT_PEPPERSPRAYED)
+
+			if(!eyes_covered)
+				to_chat(victim, "<span class='bolddanger'>Your eyes burns!</span>")
```
Comment (HMBGERDO) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm:243]:
```suggestion
				to_chat(victim, "<span class='danger'>Your eyes burns!</span>")
```

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 5]
```diff
@@ -219,56 +219,31 @@
...
+				eyes_covered = TRUE
+				safe_thing = victim.head
 			if(victim.glasses)
-				eyes_covered = 1
+				eyes_covered = TRUE
 				if(!safe_thing)
 					safe_thing = victim.glasses
-			if(eyes_covered && mouth_covered)
+
+			if(eyes_covered && victim.internal)
```
Comment (Contrabang) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm:235]: Please mention in the changelog too, that having a mask no longer counteracts pepperspray

**Review by Contrabang — CHANGES_REQUESTED** [review]

**Comment by Rurik123** [comment]
After seeing this in action against masked victims, I'll say I don't think internals requirement is for the better. Chaining into a quick baton hit for an immediate stun on anyone who doesn't have a tank on feels terrible.

I'd prefer it to be a mask check instead of an internals check. Gasmasks and such blocking it, things like cigs or the fake mustache not it.

**Comment by HMBGERDO** [comment]
alright, reverted internals check, it now checks mask as before. 
not exactly what i wanted but still better than hard stun for 6 seconds.

**Review by S34NW — APPROVED** [review]

**Comment by Dafrek** [comment]
The effects of pepper spray currently seem to go through modsuits (doesn't protect mouth or eyes) and sunglasses, (doesnt protect eyes) is this intended?

**Comment by HMBGERDO** [comment]
fixed here https://github.com/ParadiseSS13/Paradise/pull/24403
thanks for report

**Pull request merged**